packages/web-components/fast-foundation/src/avatar/avatar.ts

```diff
@@ -0,0 +1,97 @@
+import { attr, FASTElement } from "@microsoft/fast-element";
+
+/**
+ * An Avatar Custom HTML Element
+ * 
+ * @public
+ */
+export class Avatar extends FASTElement {
+  /**
+   * Indicates the Avatar should have a name.
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr public name: string;
+
+  /**
+   * Indicates the Avatar should have initials.
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: initials
+   */
+  @attr public initials: string;
+
+  /**
+   * Invoked when the `name` property changes.
+   * 
+   * @public
+   * @remarks
+   */
+  protected nameChanged(): void {
+    const result = this.name?.match(/\b\w/g) || [];
+    this.initials = ((result.shift() || '') + (result.pop() || '')).toUpperCase();
+  }
+  
+  /**
+   * Indicates the Avatar should have a color fill.
+   *  
+   * @public
+   * @remarks
+   * HTML Attribute: fill
+   */
+  @attr public fill: string;
+
+  /**
+   * Indicates the Avatar should have a text color.
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: color
+   */
+  @attr public color: string;
+
+  /**
+   * Indicates the Avatar should have an image source
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: src
+   */
+  @attr({attribute: "src"}) 
+  public imgSrc: string;
+
+  /**
+   * Indicates the Avatar should have alt text
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: alt
+   */
+  @attr public alt: string;
+
+  /**
+   * Indicates the Avatar should have url link
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: link
+   */
+  @attr public link: string;
+
+  /**
+   * Indicates the Avatar shape should be. By default it will be set to "circle".
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr public shape: string = 'circle';
```

Comment:
Perhaps we can have this be a type defined above?

```ts
type AvatarShape = "circle" | "square";

public shape: AvatarShape;
```

Last thing - we'll want to set this default in connectedCallback rather than here. In certain scenarios this will throw an exception.

Reply:
@chrisdholt I set the default like this, but it isn't working properly:

```js
  public connectedCallback(): void {
    super.connectedCallback();
    this.shape = "circle";
  }
```
The above is taking precedence over any value passed into it (i.e. Square doesn't work). Do you know what the issue might be?

Reply:
Have you removed the default from line 91?

```suggestion
  @attr public shape: string;
```

Reply:
Otherwise shape is always circle in connected callback